modules/031-linstor/docs/CONFIGURATION.md

```diff
@@ -98,3 +98,249 @@ In case your application does not support high availability and runs in a single
+
+## Backup on S3 Storage
+
+Data backup is implemented using [volume snapshots](https://kubernetes.io/docs/concepts/storage/volume-snapshots/). Snapshots are supported by the [snapshot-controller](../045-snapshot-controller/) module, which is enabled automatically for supported CSI drivers in Kubernetes clusters versions 1.20 and higher.
+
+> It works with [Amazon S3](https://aws.amazon.com/s3/) and [MinIO](https://min.io). There may be problems when working with other services.  
+>
+> Snapshots are supported only for LVMThin pools.
+
+### Creating a backup
+
+To create a snapshot of a volume and upload it to S3, follow these steps:
+1. Select (or create) `PersistentVolumeClaim`, whose data needs to be copied.
+
+   Example of `PersistentVolumeClaim`, which will be used in the examples below:
+
+   ```yaml
+   apiVersion: v1
+   kind: PersistentVolumeClaim
+   metadata:
+     name: my-linstor-volume
+     namespace: storage
+   spec:
+     accessModes:
+     - ReadWriteOnce
+     storageClassName: linstor-thindata-r2   # The name of the StorageClass.
+     resources:
+       requests:
+         storage: 2Gi
+   ```
+
+1. Create `VolumeSnapshotClass` and `Secret`, containing the access key and secret key of S3 storage.
+
+   > VolumeSnapshotClass is a cluster-wide resource. The same VolumeSnapshotClass can be used to create backups of different PVCs from different namespaces.
+
+   Example of `VolumeSnapshotClass` and `Secret`:
+
+   ```yaml
+   kind: VolumeSnapshotClass
+   apiVersion: snapshot.storage.k8s.io/v1
+   metadata:
+     name: linstor-csi-snapshot-class-s3
+   driver: linstor.csi.linbit.com
+   deletionPolicy: Retain
+   parameters:
+     snap.linstor.csi.linbit.com/type: S3
+     snap.linstor.csi.linbit.com/remote-name: backup-remote               # Linstor remote name.   
+     snap.linstor.csi.linbit.com/allow-incremental: "false"               # Whether to use incremental copies. 
+     snap.linstor.csi.linbit.com/s3-bucket: snapshot-bucket               # The name of the S3 bucket, for data storage.
+     snap.linstor.csi.linbit.com/s3-endpoint: s3.us-west-1.amazonaws.com  # S3 endpoint URL.
+     snap.linstor.csi.linbit.com/s3-signing-region: us-west-1             # The name of the S3 region. 
+     # Use virtual hosted–style or path-style S3 URL. 
+     # https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html
+     snap.linstor.csi.linbit.com/s3-use-path-style: "false"    
+     # Secret, containing the access key and secret key of access to S3 bucket.
+     csi.storage.k8s.io/snapshotter-secret-name: linstor-csi-s3-access
+     csi.storage.k8s.io/snapshotter-secret-namespace: storage
+   ---
+   kind: Secret
+   apiVersion: v1
+   metadata:
+     name: linstor-csi-s3-access
+     namespace: storage
+   immutable: true
+   type: linstor.csi.linbit.com/s3-credentials.v1
+   stringData:
+     access-key: *!ACCESS_KEY*  # S3 access key.
+     secret-key: *!SECRET_KEY*  # S3 secret key.
+   ```
```

Comment:
```suggestion
To create a snapshot of a volume and upload it to S3, follow these steps:
1. Create `VolumeSnapshotClass` and `Secret`, containing the access key and secret key of S3 storage.

   > VolumeSnapshotClass is a cluster-wide resource. The same VolumeSnapshotClass can be used to create backups of different PVCs from different namespaces.

   Example of `VolumeSnapshotClass` and `Secret`:

   ```yaml
   kind: VolumeSnapshotClass
   apiVersion: snapshot.storage.k8s.io/v1
   metadata:
     name: linstor-csi-snapshot-class-s3
   driver: linstor.csi.linbit.com
   deletionPolicy: Retain
   parameters:
     snap.linstor.csi.linbit.com/type: S3
     snap.linstor.csi.linbit.com/remote-name: backup-remote               # Linstor remote name.   
     snap.linstor.csi.linbit.com/allow-incremental: "false"               # Whether to use incremental copies. 
     snap.linstor.csi.linbit.com/s3-bucket: snapshot-bucket               # The name of the S3 bucket, for data storage.
     snap.linstor.csi.linbit.com/s3-endpoint: s3.us-west-1.amazonaws.com  # S3 endpoint URL.
     snap.linstor.csi.linbit.com/s3-signing-region: us-west-1             # The name of the S3 region. 
     # Use virtual hosted–style or path-style S3 URL. 
     # https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html
     snap.linstor.csi.linbit.com/s3-use-path-style: "false"    
     # Secret, containing the access key and secret key of access to S3 bucket.
     csi.storage.k8s.io/snapshotter-secret-name: linstor-csi-s3-access
     csi.storage.k8s.io/snapshotter-secret-namespace: storage
   ---
   kind: Secret
   apiVersion: v1
   metadata:
     name: linstor-csi-s3-access
     namespace: storage
   immutable: true
   type: linstor.csi.linbit.com/s3-credentials.v1
   stringData:
     access-key: *!ACCESS_KEY*  # S3 access key.
     secret-key: *!SECRET_KEY*  # S3 secret key.
   ```

1. Select (or create) `PersistentVolumeClaim`, whose data needs to be copied.

   Example of `PersistentVolumeClaim`, which will be used in the examples below:

   ```yaml
   apiVersion: v1
   kind: PersistentVolumeClaim
   metadata:
     name: my-linstor-volume
     namespace: storage
   spec:
     accessModes:
     - ReadWriteOnce
     storageClassName: linstor-thindata-r2   # The name of the StorageClass.
     resources:
       requests:
         storage: 2Gi
   ```
```